.github/workflows/ci.yml

```diff
@@ -0,0 +1,141 @@
+        with:
+          maven-version: 3.2.5
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Compile flink-connector-elasticsearch
+        run: mvn ${{ env.MVN_GOALS }} ${{ env.MVN_PROFILES }} ${{ env.MVN_TEST_OPTIONS }} ${{ env.MVN_COMMON_OPTIONS }} ${{ env.MVN_CONNECTION_OPTIONS }}
+
+  test_ci_jdk8:
+    needs: compile_ci_jdk8
+    runs-on: ubuntu-latest
+    env:
+      MVN_GOALS: clean install
+      MVN_PROFILES: -Dscala-2.12
+      MVN_TEST_OPTIONS: -B --no-snapshot-updates
+      MVN_COMMON_OPTIONS: -Dflink.forkCount=2 -Dflink.forkCountTestPackage=2
+      MVN_CONNECTION_OPTIONS: -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.httpconnectionManager.ttlSeconds=120
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Test - connectors
+        run: mvn ${{ env.MVN_GOALS }} ${{ env.MVN_PROFILES }} ${{ env.MVN_TEST_OPTIONS }} ${{ env.MVN_COMMON_OPTIONS }} ${{ env.MVN_CONNECTION_OPTIONS }}
+
+  compile_ci_jdk11:
+    runs-on: ubuntu-latest
+    env:
+      MVN_GOALS: clean install
+      MVN_PROFILES: -Pcheck-convergence -Dscala-2.12
+      MVN_TEST_OPTIONS: -Dmaven.javadoc.skip=true -DskipTests -Dfast -U -B
+      MVN_COMMON_OPTIONS: -Dflink.forkCount=2 -Dflink.forkCountTestPackage=2
+      #      Temporary disable convergence phase since that will fail
+      #      MVN_COMMON_OPTIONS: -Dflink.convergence.phase=install -Dflink.forkCount=2 -Dflink.forkCountTestPackage=2
+      MVN_CONNECTION_OPTIONS: -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.httpconnectionManager.ttlSeconds=120
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+
+      - name: Set JDK
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+
+      - name: Set Maven 3.2.5
+        uses: stCarolas/setup-maven@v4.2
+        with:
+          maven-version: 3.2.5
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Compile flink-connector-elasticsearch
+        run: mvn ${{ env.MVN_GOALS }} ${{ env.MVN_PROFILES }} ${{ env.MVN_TEST_OPTIONS }} ${{ env.MVN_COMMON_OPTIONS }} ${{ env.MVN_CONNECTION_OPTIONS }}
+
+  test_ci_jdk11:
+    needs: compile_ci_jdk11
```

Comment:
jdk switches could be easily modeled as a matrix.

this job is also not using the correct jdk nor maven version.